### ISSUE TITLE
Renovate doesn't always update yarn.lock, which makes CI fail

### ISSUE THREAD

**Issue by laurent22** [renovatebot/renovate#18801]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

In some cases, Renovate doesn't update yarn.lock, and that makes CI fail.

Here's an example:

https://github.com/laurent22/joplin/pull/7117

And the CI log file would look like this (just an extract because it's very long):

	22-11-07T14:02:13.2274610Z ➤ YN0000: │ @@ -35284,9 +35303,9 @@
	2022-11-07T14:02:13.2275880Z ➤ YN0000: │ checksum: 4b765efcd5049994a051e91af81d35bbfdba0ed8ef2ca882a7c955528526eb446520516615a70388be764929d2ab7a68e22a1062fa2e9c94ff7b13758b198ee6
	2022-11-07T14:02:13.2277570Z ➤ YN0000: │ languageName: node
	2022-11-07T14:02:13.2278240Z ➤ YN0000: │ linkType: hard
	2022-11-07T14:02:13.2278850Z ➤ YN0000: │
	2022-11-07T14:02:13.2279590Z ➤ YN0028: │ -"yargs@npm:16.2.0, yargs@npm:^16.2.0":
	2022-11-07T14:02:13.2280270Z ➤ YN0028: │ +"yargs@npm:16.2.0":
	2022-11-07T14:02:13.2280990Z ➤ YN0000: │ version: 16.2.0
	2022-11-07T14:02:13.2281720Z ➤ YN0000: │ resolution: "yargs@npm:16.2.0"
	2022-11-07T14:02:13.2282400Z ➤ YN0000: │ dependencies:
	2022-11-07T14:02:13.2283060Z ➤ YN0000: │ cliui: ^7.0.2
	2022-11-07T14:02:13.2283650Z ➤ YN0000: │ @@ -35367,8 +35386,22 @@
	2022-11-07T14:02:13.2285300Z ➤ YN0000: │ checksum: 00d58a2c052937fa044834313f07910fd0a115dec5ee35919e857eeee3736b21a4eafa8264535800ba8bac312991ce785ecb8a51f4d2cc8c4676d865af1cfbde
	2022-11-07T14:02:13.2286300Z ➤ YN0000: │ languageName: node
	2022-11-07T14:02:13.2287240Z ➤ YN0000: │ linkType: hard
	2022-11-07T14:02:13.2287840Z ➤ YN0000: │
	2022-11-07T14:02:13.2288520Z ➤ YN0028: │ +"yargs@npm:^17.5.1":
	2022-11-07T14:02:13.2289170Z ➤ YN0028: │ + version: 17.6.2
	2022-11-07T14:02:13.2289890Z ➤ YN0028: │ + resolution: "yargs@npm:17.6.2"
	2022-11-07T14:02:13.2290660Z ➤ YN0028: │ + dependencies:
	2022-11-07T14:02:13.2291560Z ➤ YN0028: │ + cliui: ^8.0.1
	2022-11-07T14:02:13.2292420Z ➤ YN0028: │ + escalade: ^3.1.1
	2022-11-07T14:02:13.2293130Z ➤ YN0028: │ + get-caller-file: ^2.0.5
	2022-11-07T14:02:13.2294040Z ➤ YN0028: │ + require-directory: ^2.1.1
	2022-11-07T14:02:13.2294780Z ➤ YN0028: │ + string-width: ^4.2.3
	2022-11-07T14:02:13.2295460Z ➤ YN0028: │ + y18n: ^5.0.5
	2022-11-07T14:02:13.2296130Z ➤ YN0028: │ + yargs-parser: ^21.1.1
	2022-11-07T14:02:13.2296830Z ➤ YN0028: │ + languageName: node
	2022-11-07T14:02:13.2297460Z ➤ YN0028: │ + linkType: hard
	2022-11-07T14:02:13.2298060Z ➤ YN0028: │ +
	2022-11-07T14:02:13.2298680Z ➤ YN0000: │ "yargs@npm:^7.1.0":
	2022-11-07T14:02:13.2299300Z ➤ YN0000: │ version: 7.1.2
	2022-11-07T14:02:13.2299940Z ➤ YN0000: │ resolution: "yargs@npm:7.1.2"
	2022-11-07T14:02:13.2300700Z ➤ YN0000: │ dependencies:
	2022-11-07T14:02:13.2301230Z ➤ YN0000: │
	2022-11-07T14:02:13.2302050Z ➤ YN0028: │ The lockfile would have been modified by this install, which is explicitly forbidden.

This happens for some pull requests but all of them, so I'm not sure what's causing it.

I believe the Renovate log is this one: https://app.renovatebot.com/dashboard#github/laurent22/joplin/879433923 However I don't know what might be relevant in it. Happy to dig into it if you can let me know what to look for.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

No reproduction, but I have linked to a public repo where it occurs

**Comment by viceice** [comment]
Copy the logs when the branch was created. Only paste the lines for that branch, not the full log.

**Comment by laurent22** [comment]
> Only paste the lines for that branch, not the full log.

I found the log [here](https://app.renovatebot.com/dashboard#github/laurent22/joplin/880064650), which obviously is huge, and I'm not sure what I need to paste here?

**Comment by viceice** [comment]
Type the relevant branch name to the filter, eg `renovate/major-jest-monorepo` (you can also only use some prefix parts. then the log should be much smaller.

I've no acces to your logs until i'm a maintainer on your repo.

**Comment by laurent22** [comment]
Ok I've done that but the log is still quite large so I can't paste it in the issue. I've attached it as a file instead if that can help: https://github.com/renovatebot/renovate/files/9953452/eslint-interactive-renovate-log.txt

**Comment by viceice** [comment]
I've told you to filter for the one relevant branch. you've posted the full log, i've no time to analyze that.

**Comment by laurent22** [comment]
Ok I didn't realise I published the whole log, but then it's also not clear how to filter it. This is what I do:

- I open the log
- In "Branch", I type the pull request branch
- And then... well I don't think anything happens:

<img width="1141" alt="image" src="https://user-images.githubusercontent.com/1285584/200379755-4d3e7337-7344-4122-ab03-419568947c07.png">

If the above steps are not correct, please could you let me know how to get branch filtering working?

**Comment by laurent22** [comment]
For info, if I only display the warnings in this log, it shows this:

```
WARN: artifactErrors(branch="renovate/eslint-interactive-10.x")
{
  "artifactErrors": [
    {
      "lockFile": "yarn.lock",
      "stderr": "bash: yarn: command not found\n"
    }
  ]
}
WARN: artifactErrors(branch="renovate/eslint-plugin-promise-6.x")
{
  "artifactErrors": [
    {
      "lockFile": "yarn.lock",
      "stderr": "bash: yarn: command not found\n"
    }
  ]
}
```

Sounds like it could be related?

**Comment by viceice** [comment]
for me the branch filter works expected. be sure to not include any quotes or spaces

**Comment by laurent22** [comment]
But unfortunately I'm not able to filter by branch, as described above. Entering anything in the branch field simply doesn't do anything.

I see that currently it's the packages that are installed at the root of the monorepo that are having this problem. Anything under `/packages` is added properly, and yarn.lock is updated, but anything at the root is not. In this case, it updates package.json, but doesn't update yarn.lock (and maybe fails to run yarn at all, as the above error shows).

Is that a known bug by any chance? And any idea how it can be fixed?

**Comment by viceice** [comment]
![Screenshot_20221107-195925.png](https://user-images.githubusercontent.com/1798109/200392527-98d6956e-bb17-4ca8-9109-caa2c939ef36.png)

**Comment by viceice** [comment]
do you have any script blocker in your browser? the filter needs JavaScript of cause

**Comment by laurent22** [comment]
Yes I have JS enabled, and to be sure I disabled uBlock but still the same. No error in the console either. In fact if I enter a random branch name, as below, it still shows the whole log so filtering doesn't seem to be working in my case.


<img width="484" alt="image" src="https://user-images.githubusercontent.com/1285584/200393215-8a84b005-0d05-4936-b199-e9819292d95f.png">

**Comment by rarkins** [comment]
Filter manually instead

**Comment by laurent22** [comment]
Ok I've grepped the log and extracted the parts related to the PR branch, it's now in the top post. Any what might be the issue?

**Comment by viceice** [comment]
> Ok I've grepped the log and extracted the parts related to the PR branch, it's now in the top post. Any idea what might be the issue?

Please don't remove the logs summary section next time, so logs are collapsed by default
